### PR TITLE
[git-webkit] Improve user prompts (Part 1)

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 3d81614767e44c539afb999b51540b058462f051
<pre>
[git-webkit] Improve user prompts (Part 5)
<a href="https://bugs.webkit.org/show_bug.cgi?id=235655">https://bugs.webkit.org/show_bug.cgi?id=235655</a>
&lt;rdar://problem/88082697 &gt;

Reviewed by Aakash Jain and Dewei Zhu.

* Tools/Scripts/git-webkit: Log to 'info' not 'warning.'

Canonical link: https://commits.webkit.org/246698@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@288969 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>
----------------------------------------------------------------------
#### 9de7e660755778ae0834a5176351eb86b39d6658
<pre>
[git-webkit] Improve user prompts (Part 4)
<a href="https://bugs.webkit.org/show_bug.cgi?id=235655">https://bugs.webkit.org/show_bug.cgi?id=235655</a>
&lt;rdar://problem/88082697 &gt;

Reviewed by Aakash Jain and Dewei Zhu.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.main): Make it clear why we're prompting the user for a branch name.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
(TestBranch.test_prompt_git):

Canonical link: https://commits.webkit.org/246697@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@288968 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>
----------------------------------------------------------------------
#### 6484f5aeb96f850a123d8626fcf91b3f57d61e7b
<pre>
[git-webkit] Improve user prompts (Part 3)
<a href="https://bugs.webkit.org/show_bug.cgi?id=235655">https://bugs.webkit.org/show_bug.cgi?id=235655</a>
&lt;rdar://problem/88082697 &gt;

Reviewed by Aakash Jain and Dewei Zhu.

* Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
(Setup.git): Make it clear to users that configurations are being set only
on the local repository.
* Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py:

Canonical link: https://commits.webkit.org/246696@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@288967 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>
----------------------------------------------------------------------
#### 1632b86d3e22d3e39dbd238fc3e1f166aee3d5d7
<pre>
[git-webkit] Improve user prompts (Part 2)
<a href="https://bugs.webkit.org/show_bug.cgi?id=235655">https://bugs.webkit.org/show_bug.cgi?id=235655</a>
&lt;rdar://problem/88082697 &gt;

Reviewed by Aakash Jain and Dewei Zhu.

* Tools/Scripts/libraries/webkitcorepy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py:
(Terminal.input): Gracefully handle keyboard interrupt
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/tests/terminal_unittest.py:
(TerminalTests.test_interrupt):

Canonical link: https://commits.webkit.org/246695@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@288966 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>
----------------------------------------------------------------------
#### 1632b86d3e22d3e39dbd238fc3e1f166aee3d5d7
<pre>
[git-webkit] Improve user prompts (Part 1)
<a href="https://bugs.webkit.org/show_bug.cgi?id=235655">https://bugs.webkit.org/show_bug.cgi?id=235655</a>
&lt;rdar://problem/88082697 &gt;

Reviewed by Aakash Jain and Dewei Zhu.

* Tools/Scripts/libraries/webkitbugspy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py: Use token form url.
* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py: Use token form url.

Canonical link: https://commits.webkit.org/246694@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@288965 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

**Comment by JonWBedard** [comment]
I wanted to have a practical example of a patch-series to discuss.

**Review by dewei-zhu — APPROVED** [review]
r=me

File: Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py [hunk 1]
```diff
@@ -36,7 +36,11 @@ class Terminal(object):
 
     @classmethod
     def input(cls, *args, **kwargs):
-        return (input if sys.version_info > (3, 0) else raw_input)(*args, **kwargs)
+        try:
+            return (input if sys.version_info > (3, 0) else raw_input)(*args, **kwargs)
+        except KeyboardInterrupt:
+            sys.stderr.write('\nUser interrupted program when prompted for input\n')
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py]: Nit: do we need to write this explicitly (especially `when prompted for input`). If user pressed ctrl+c, they know what they did 😆

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def main(cls, args, repository, **kwargs):
             return 1
 
         if not args.issue:
-            args.issue = Terminal.input('Branch name: ')
+            args.issue = Terminal.input('Name of branch to create: ')
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py]: Separate question, we don't really use this name as branch name, we prepend `eng/` to it, right?

File: Tools/Scripts/git-webkit [hunk 3]
```diff
@@ -50,7 +50,7 @@ def additional_setup(args, repository):
     if not isinstance(repository, local.Git):
         return 0
 
-    log.warning('Setting merging behavior for changelogs...')
+    log.info('Setting merging behavior for changelogs...')
```
Comment (aj062) [inline Tools/Scripts/git-webkit]: would it prevent it from being printed in regular `git-webkit setup` workflow?

**Review by aj062 — APPROVED** [review]

File: Tools/Scripts/git-webkit (hunk 3 above)
Comment (JonWBedard) [reply Tools/Scripts/git-webkit]: Yes! I forgot to change this in a change a few weeks back.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py (hunk 2 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py]: Unless the user prepends `eng/` themselves, yes.

@aproskuryakov had indicated he though "Branch name" wasn't specific enough, hence the change. Not sure what the right prompt here is.

File: Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py]: I can ditch the `when prompted for input` part. I think the `User interrupted program` is helpful, if nothing else, for unit testing.

Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py]: Small side note: will be updating this PR before landing to address this feedback, and to see what happens when addressing feedback form a commit series like this.

**Comment by JonWBedard** [comment]
@aproskuryakov requested pre-filling the branch prompt with the `eng` prefix. This does seem to be possible, in certain terminals, but also looks to be involved enough that it should be a different pull-request.

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/246698@main (3d81614767e44c539afb999b51540b058462f051)!

**Pull request merged**